helm/dagger/templates/engine-daemonset.yaml

```diff
@@ -0,0 +1,89 @@
+---
+{{- if .Values.magicache.enabled }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "dagger.fullname" . }}-magicache-token
+  namespace:  {{ .Release.Namespace }}
+data:
+  _EXPERIMENTAL_DAGGER_CACHESERVICE_TOKEN: {{ required "A magicache token is required" .Values.magicache.token | b64enc | quote }}
+{{- end }}
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "dagger.fullname" . }}-engine
+  namespace: {{ .Release.Namespace }}
+  labels:
+    k8s-app: dagger-engine
+spec:
+  selector:
+    matchLabels:
+      name: dagger-engine
+  template:
+    metadata:
+      labels:
+        name: dagger-engine
+    spec:
+      {{- with .Values.engine.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.engine.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        runAsUser: 0
+        runAsGroup: 1001
+        fsGroup: 1001
+        fsGroupChangePolicy: "OnRootMismatch"
+      initContainers:
+        - name: download-bin
+          image: {{ .Values.engine.image.repository }}:{{ if .Values.engine.image.tag }}{{ .Values.engine.image.tag }}{{ else }}{{ .Chart.AppVersion }}{{ end }}
+          command: ["sh", "-c", "wget https://github.com/moby/buildkit/releases/download/v0.11.6/buildkit-v0.11.6.linux-amd64.tar.gz -O /shared-bin/buildkit.tar.gz && tar -xvzf /shared-bin/buildkit.tar.gz --strip-components 1 bin/buildctl -C /shared-bin"]
+          volumeMounts:
+          - name: shared-bin
+            mountPath: /shared-bin
+      containers:
+        - name: dagger-engine
+          image: {{ .Values.engine.image.repository }}:{{ if .Values.engine.image.tag }}{{ .Values.engine.image.tag }}{{ else }}{{ .Chart.AppVersion }}{{ end }}
+          imagePullPolicy: Always
+          args:
+            - "--oci-max-parallelism"
+            - "num-cpu"
+          {{- if .Values.magicache.enabled }}
+          env:
+          - name: _EXPERIMENTAL_DAGGER_CACHESERVICE_URL
+            value: {{ required "A magicache url is required" .Values.magicache.url }}
+          envFrom:
+          - secretRef:
+              name: {{ include "dagger.fullname" . }}-magicache-token
+          {{- end }}
+          securityContext:
+            privileged: true
+            capabilities:
+              add:
+                - ALL
+          readinessProbe:
+            exec:
+              command: ["/shared-bin/buildctl", "debug", "workers"]
+            initialDelaySeconds: 5
+            periodSeconds: 5
+          volumeMounts:
+            - name: varlibdagger
+              mountPath: /var/lib/dagger
+            - name: varrundagger
+              mountPath: /var/run/buildkit
+            - name: shared-bin
+              mountPath: /shared-bin
+      terminationGracePeriodSeconds: 300
+      volumes:
```

Comment:
I suggest exposing an engine config for people who want to configure the engine via a configmap

Something like this:

```yaml
          volumeMounts:
            - name: varlibdagger
              mountPath: /var/lib/dagger
            - name: varrundagger
              mountPath: /var/run/buildkit
            - name: dagger-engine-config
              mountPath: /etc/dagger/engine.toml
              subPath: engine.toml
      volumes:
        - name: varlibdagger
          hostPath:
            path: /var/lib/dagger
        - name: varrundagger
          hostPath:
            path: /var/run/dagger
        - name: dagger-engine-config
          configMap:
            name: dagger-engine-config
            items:
              - key: engine.toml
                path: engine.toml
```

Where your engine.toml in the configmap might look something like

```yaml
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: dagger-engine-config
  namespace: actions-runner-system
data:
  engine.toml: |
    debug = true
    insecure-entitlements = ["security.insecure"]
    [registry."registry:5000"]
      http = true
```

Not that I suggest using these values, of course, only that we should expose the ability to mount an engine config by default as it's a pretty common scenario